### DISCUSSION TITLE
Curious about iterations

### DISCUSSION THREAD

**Discussion by jazzmonger** [AzonInc/Doorman#34] [General]
Nice, clean design. I appreciate you sharing all your KiCad files.  Not that many good examples out there of something we can all take and build on.  Questions:

why 4 layers?  Seems like you could do it w/ 2?

How many board iterations did you go through before you finished it?

Nice idea on the solid state relay! folks often forget about those.

Have any other designs that have USB support chips like CP21xx?  I have a design that uses 3 full UARTs and only the ESP32 original WROOM style chips (not S2/S3/C2/C3 etc variations) support that.

**Comment by AzonInc** [comment]
Thank you :)
I'm happy that people like it and I hope that there are some people that can contribute to the hardware design to make it as good as the competitors but still open source. Or maybe it just helps people to build something else based on this example project.

> why 4 layers? Seems like you could do it w/ 2?

It didn't make a difference in the final price and was easier to route but yea in general its possible to have it on 2-layers only.

> How many board iterations did you go through before you finished it?

The first manufactured revision was actually working just fine and didnt need any further changes, however there were some changes recently to remove some noise in the indoor station audio while still allowing to power the device directly via 2 bus wires.
The first versions never got manufactured, it started with 1.3, 1.4 added one diode only and 1.5 had the power supply changes.
Every manufactured revision is also tagged as a release in the reposirory.

It all started on a breadboard here: https://github.com/peteh/doorman/issues/17

> Nice idea on the solid state relay! folks often forget about those.

I tried to make the device as versatile as possible to also support some older systems in case people move to a new place. It would be a useless device if you move to a new place which doesn't have a TCS/Koch or similar intercom.

> Have any other designs that have USB support chips like CP21xx? I have a design that uses 3 full UARTs and only the ESP32 original WROOM style chips (not S2/S3/C2/C3 etc variations) support that.

At first I tried something with the CP21xx but I discarded that in favor of the ESP32-S3's USB support.


If you have any suggestions for improving the design, I would be very interested ofc :)

**Reply by jazzmonger** [reply]
Thanks for the info.  I'm already diving into the KiCad files.

**Reply by AzonInc** [reply]
And here u see the beginning of everything: https://github.com/peteh/doorman/issues/17

**Reply by AzonInc** [reply]
Also the last experiment was using the esp chip itself - see dev-esp-small branch

Maybe it could make some space on the PCB and add new possibilities for working with audio.

Something as small as the supermini would be saving some space I guess.

![Screenshot_2024-11-25-21-42-38-75_40deb401b9ffe8e1df2f1cc5ba480b12.jpg](https://github.com/user-attachments/assets/b18d2512-6efe-4f05-923f-27858c4dd4aa)

**Reply by jazzmonger** [reply]
so, that's whats inside the C3 module?

**Reply by AzonInc** [reply]
Yea basically but I'd stick to the s3
